esp-hal-common/src/spi.rs

```diff
@@ -1012,12 +1050,42 @@ where
         let reg_block = self.register_block();
         self.configure_datalen(len as u32 * 8);
 
+        self.enable_dma();
+        self.update();
+
+        rx.prepare_transfer(self.dma_peripheral(), ptr, len)?;
+
+        self.clear_dma_interrupts();
+
+        reg_block.cmd.modify(|_, w| w.usr().set_bit());
+
+        return Ok(());
+    }
+
+    fn dma_peripheral(&self) -> DmaPeripheral {
+        match self.spi_num() {
+            2 => DmaPeripheral::Spi2,
+            #[cfg(esp32)]
+            3 => DmaPeripheral::Spi3,
+            _ => panic!("Illegal SPI instance"),
+        }
+    }
+
+    #[cfg(esp32c3)]
+    fn enable_dma(&self) {
+        let reg_block = self.register_block();
         reg_block.dma_conf.modify(|_, w| w.dma_tx_ena().set_bit());
         reg_block.dma_conf.modify(|_, w| w.dma_rx_ena().set_bit());
-        self.update();
+    }
 
-        rx.prepare_transfer(crate::dma::gdma::DmaPeripheral::Spi2, ptr, len)?;
+    #[cfg(esp32)]
+    fn enable_dma(&self) {
+        // for non GDMA this is done in `assign_tx_device` / `assign_rx_device`
```

Comment:
Hi @bjoernQ,
Can you clarify this line, please?
Where is exactly DMA enabled for ESP32?

Reply:
On ESP32/ESP32-S2 DMA works very different from other chips. On other chips there is a dedicated peripheral (GDMA) while on ESP32/ESP32-S2 the DMA channel registers are part of the peripheral using it (and are dedicated to that one peripheral)

So, on the "older" chips we just start the DMA transfer and the peripheral knows it